### ISSUE TITLE
Migrate service tweaks to Homebrew

### ISSUE THREAD

**Issue by nstanger** [nstanger/tcsh_initialisation#3]
In particular `set.mariadb.maxfiles.plist` and `set.postgresql.netrestart.plist` as these are currently coded for MacPorts.

Note that the `plist` files for brew services are installed in `~/Library/LaunchAgents` rather than `/Library/LaunchDaemons` unless you use `sudo` (which is probably what I actually want).

`brew services restart mariadb@10.2`, but does that remove and re-install the `.plist` file? (Problematic if so as we needs to edit it to increase the `maxfiles` limit.)

**Comment by nstanger** [comment]
Interesting, MariaDB on Big Sur has `@@open_files_limit` = 32190 despite the usual system-level setting `maxfiles` = 256 and not running the `set.mariadb.maxfiles` agent. Maybe this is no longer an issue under Big Sur? We’ll find out when I upgrade `sobmac0011`.

Remote login to PostgreSQL is still an issue, although that was only needed for a couple of things that connected directly to the database, like the Tcl/Tk marking form. This is bad practice, and under INFORMS3 these kinds of things will be migrated to the REST API.

So in summary, it may not be necessary to migrate these services at all. Certainly low priority for now.

**Comment by nstanger** [comment]
OK, that’s annoying: `brew services restart` does in fact completely replace the `.plist` file.

Starting the service with `sudo` does indeed put the `.plist` into `/Library/LaunchDaemons`, but then I can’t connect to the database:

```
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/tmp/mysql.sock' (2)
```

Probably the permissions on the database directory are all wrong (and Homebrew does change ownership of a few things to `root`). Regardless, this doesn’t stop the `restart` from replacing the `.plist`.

**Comment by nstanger** [comment]
Editing the `.plist` followed by:

```sh
launchctl unload homebrew.mxcl.mariadb@10.2.plist
launchctl load homebrew.mxcl.mariadb@10.2.plist
```

does work, so this is the fallback position if Big Sur doesn’t just fix the problem.

**Comment by nstanger** [comment]
Yep, MariaDB seems to just automatically start up with ``@@open_files_limit` > 32000. Problem solved for MariaDB.

Won’t fix for PostgreSQL for the reasons outlined above.
